### PR TITLE
Add patterns for building a jenkins job

### PR DESCRIPTION
**Pull request by jzoldak**
@edx/platform-team 
@cpennington FYI

### CODE REVIEW

File: edxpipelines/patterns/stages.py [hunk 1]
```diff
@@ -967,3 +967,34 @@ def generate_armed_stage(pipeline, stage_name):
...
+        gomatic.Stage
+    """
+    jenkins_build_stage = pipeline.ensure_stage(stage_name)
+    jenkins_build_job = jenkins_build_stage.ensure_job(build_job_name)
+    tasks.generate_requirements_install(trigger_job, 'tubular')
+    trigger_job.add_task(ExecTask(
+        [
+            '/bin/bash',
+            '-c',
+            '/usr/bin/python scripts/jenkins_trigger_build.py --url "${JENKINS_URL}" --user_name "${JENKINS_USER_NAME}" --job "${JENKINS_JOB_NAME}" --cause "Triggered by GoCD Pipeline ${GO_PIPELINE_NAME} build ${GO_PIPELINE_LABEL}" --param ${JENKINS_PARAM}',
```
Comment (jzoldak) [inline edxpipelines/patterns/stages.py]: @macdiesel can you pls help me sort out how/where to handle the environment variables / parameters?

File: edxpipelines/pipelines/manual_verification.py [hunk 2]
```diff
@@ -104,6 +104,24 @@ def install_pipeline(save_config_locally, dry_run, variable_files, cmd_line_vars
...
+        pipeline_job_name = jenkins.get('pipeline_job_name')
+        jenkins_url = jenkins.get('url')
+        jenkins_user_name = jenkins.get('user_name')
+        jenkins_job_name = jenkins.get('job_name')
+        jenkins_param = jenkins.get('param')
+
+        pipeline.ensure_unencrypted_secure_environment_variables(
+            {
+                'JENKINS_USER_TOKEN': jenkins.get('user_token'),
+                'JENKINS_JOB_TOKEN': jenkins.get('job_token')
```
Comment (macdiesel) [inline edxpipelines/pipelines/manual_verification.py]: Are the user and job tokens different for each job? If they are you will need to store them under different environment variable names.

File: edxpipelines/pipelines/manual_verification.py [hunk 3]
```diff
@@ -104,6 +104,24 @@ def install_pipeline(save_config_locally, dry_run, variable_files, cmd_line_vars
         )
     )
 
+    for jenkins in config['jenkins_verifications']:
```
Comment (macdiesel) [inline edxpipelines/pipelines/manual_verification.py]: Please put these stages before the manual_verification_stage.

File: edxpipelines/pipelines/manual_verification.py [hunk 4]
```diff
@@ -104,6 +104,24 @@ def install_pipeline(save_config_locally, dry_run, variable_files, cmd_line_vars
...
+        jenkins_param = jenkins.get('param')
+
+        pipeline.ensure_unencrypted_secure_environment_variables(
+            {
+                'JENKINS_USER_TOKEN': jenkins.get('user_token'),
+                'JENKINS_JOB_TOKEN': jenkins.get('job_token')
+            }
+        )
+
+        job = jenkins_stage.ensure_job(pipeline_job_name)
```
Comment (macdiesel) [inline edxpipelines/pipelines/manual_verification.py]: Where is jenkins_stage defined?

Comment (macdiesel) [inline edxpipelines/pipelines/manual_verification.py]: Also the jenkins stage should be set to has_manual_approval()

File: edxpipelines/pipelines/manual_verification.py [hunk 5]
```diff
@@ -91,6 +91,37 @@ def install_pipeline(save_config_locally, dry_run, variable_files, cmd_line_vars
...
+    )
+
+    # Create the stage with the Jenkins jobs
+    jenkins_stage = pipeline.ensure_stage(constants.JENKINS_VERIFICATION_STAGE_NAME)
+    jenkins_stage.set_has_manual_approval()
+
+    for jenkins in config['jenkins_verifications']:
+        pipeline_job_name = jenkins['pipeline_job_name']
+        jenkins_url = jenkins['url']
+        jenkins_user_name = jenkins['user_name']
```
Comment (doctoryes) [inline edxpipelines/pipelines/manual_verification.py]: This should be `config['jenkins_user_name']`.

File: edxpipelines/pipelines/manual_verification.py [hunk 6]
```diff
@@ -91,6 +91,37 @@ def install_pipeline(save_config_locally, dry_run, variable_files, cmd_line_vars
...
+        {
+            'JENKINS_USER_TOKEN': config['jenkins_user_token'],
+            'JENKINS_JOB_TOKEN': config['jenkins_job_token']
+        }
+    )
+    pipeline.ensure_environment_variables(
+        {
+            'JENKINS_USER_NAME': config['jenkins_user_name'],
+        }
+    )
```
Comment (doctoryes) [inline edxpipelines/pipelines/manual_verification.py]: Is there a need for this environment variable? Does it get used anywhere? Looks like the username is explicitly passed in below and hard-coded into the task.

**Pull request merged**